packages/utils/test/route.test.js

```diff
@@ -238,4 +238,109 @@ describe('util: route', () => {
       expect(routesResult).toMatchSnapshot()
     })
   })
+
+  describe('util: route sortRoutes', () => {
+    const files = [
+      'pages/_param.vue',
+      'pages/de/index.vue',
+      'pages/index.vue',
+      'pages/*.vue',
```

Comment:
shouldn't this (and other occurrences) be `_.vue` instead?

Reply:
If I'm not mistaken, nuxt uses the `_.vue` as a catch-all, right ?
I changed here and when using `_` it goes back to the initial problem.
I found out the [getRoutePathExtension](https://github.com/nuxt/nuxt/blob/696dbc68447c95527c0d5dc2c324d5f93e4dc0cf/packages/utils/src/route.js#LL243C7-L243C28) method, and I am adding a snippet there to check the `_` in the end of the key. This should fix the problem for this use case.

Reply:
Actually I think you should not be doing anything with `_` that is neither the only character nor followed by some text.

You can test it yourself by creating `pages/test_.vue` file. Nuxt doesn't convert the `_` to `*`. So it's supposed to be handled as literal `_`.

Reply:
Hey @rchl, 
I tested and if I add the route `de_.vue` we go back to the initial problem. The `de_` route goes to the top of the list and catches all `de` routes. 
I changed the [getRoutePathExtension](https://github.com/nuxt/nuxt/blob/d24275a1dc85b5be29a49cc23b66a1eab2f94552/packages/utils/src/route.js#L243) to handle the `_` in the end of the string (now it converts to *) and with that change everything seems to be right.

Reply:
Just test manually by creating a `de_.vue` file. Nuxt generates this route:
```
{
    path: "/de_",
    component: _0b861482,
    name: "de_"
  }
```

where `_` is treated literally and not converted to `*`. So it's fine if it's at the beginning and it shouldn't cause any issues because it's a literal match.

Reply:
Ok, I understood now. 
My only question now is: Should the `_` in the end of the value be treated as a literal or not ?
For me, it makes sense to be treated as a special character, but I'm not sure if Nuxt should handle it like that, either way seems fine to me.
@danielroe can you give your insights on it ?

Reply:
[File-based routing in Nuxt](https://v2.nuxt.com/docs/features/file-system-routing) only supports the case when key is equal `_` or is followed by some characters. But modules can add routes with `*` anywhere in fact and the user can also do it using the [extendRoutes](https://v2.nuxt.com/docs/features/file-system-routing#extendroutes) option. So sorting of those should IMO still be supported but the changes should not affect the capabilities of the file-based routing (and I think your change in `getRoutePathExtension` would affect those).

Reply:
Yes, it affects those. I'll roll back the commit with changes on the `getRoutePathExtension`